### PR TITLE
RUN-3460: iframes not getting preload scripts

### PR DESCRIPTION
**Pull request by wenjunche**
Test results
[W10](https://testing-dashboard.openfin.co/#/app/sessions/api/completed/59e8068aaf26a25be2ffc903)
[W7](https://testing-dashboard.openfin.co/#/app/sessions/api/completed/59e8060faf26a25be2ffc902)

### CODE REVIEW

**Comment by whyn07m3** [comment]
New broken test: "Iframe steals connection"?

**Comment by wenjunche** [comment]
that test should fail with '--framestrategy=frames'

**Comment by wenjunche** [comment]
Added more changes to fix errors with updating preload states of iframes

Test results
[W10](https://testing-dashboard.openfin.co/#/app/sessions/api/completed/59e8fd85af26a25be2ffc906)
[W7](https://testing-dashboard.openfin.co/#/app/sessions/api/completed/59e8fd1aaf26a25be2ffc905)

**Comment by whyn07m3** [comment]
So this needs js docs update, because it's adding a new property "framePreloadState"

**Comment by wenjunche** [comment]
Merged with develop

Test results:
[W10](https://testing-dashboard.openfin.co/#/app/sessions/api/completed/59ef9ee4af26a25be2ffc94b)
[W7](https://testing-dashboard.openfin.co/#/app/sessions/api/completed/59ef9ea7af26a25be2ffc94a)

File: src/browser/api/window.js [hunk 1]
```diff
@@ -1164,15 +1165,48 @@ Window.setWindowPreloadState = function(identity, payload) {
...
+    const frameInfo = coreState.getInfoByUuidFrame(identity);
+    let openfinWindow;
+    if (frameInfo.entityType === 'iframe') {
+        openfinWindow = Window.wrap(frameInfo.parent.uuid, frameInfo.parent.name);
+    } else {
+        openfinWindow = Window.wrap(uuid, name);
     }
 
-    ofEvents.emit(route.window(updateTopic, uuid, name), { name, uuid, preloadState });
+    if (openfinWindow) {
```
Comment (whyn07m3) [inline src/browser/api/window.js]: can this be

```
if (!openfinWindow) {
   return log.writeToLog('info', `setWindowPreloadState missing openfinWindow ${uuid} ${name}`);
}
```

so that we don't nest the whole block of code with a tab?

File: src/browser/api/window.js [hunk 2]
```diff
@@ -1098,7 +1100,6 @@ Window.getWindowInfo = function(identity) {
         title: webContents.getTitle(),
         url: webContents.getURL()
     };
-
```
Comment (whyn07m3) [inline src/browser/api/window.js]: why remove this nice beautiful line? :)

File: src/browser/api/window.js [hunk 3]
```diff
@@ -1088,7 +1089,8 @@ Window.getGroup = function(identity) {
 
 Window.getWindowInfo = function(identity) {
     const browserWindow = getElectronBrowserWindow(identity, 'get info for');
-    const { pluginState, preloadState } = Window.wrap(identity.uuid, identity.name);
+    const openfinWindow = Window.wrap(identity.uuid, identity.name);
+    const { pluginState, preloadState } = openfinWindow;
```
Comment (whyn07m3) [inline src/browser/api/window.js]: why this change (from 1 to 2 lines)?

**Comment by wenjunche** [comment]
Updated based on feedbacks

[W10](https://testing-dashboard.openfin.co/#/app/sessions/api/completed/59f0a2b9af26a25be2ffc955)
[W7](https://testing-dashboard.openfin.co/#/app/sessions/api/completed/59f0a268af26a25be2ffc954)

**Comment by wenjunche** [comment]
Created RUN-3507
